x/feedist/handler_test.go

```diff
@@ -47,17 +87,212 @@ func TestHandlers(t *testing.T) {
+							// All funds should be send to this account.
+							{Weight: 123456789, Address: addr1},
+						},
+					},
+					blocksize:      100,
+					wantCheckErr:   nil,
+					wantDeliverErr: nil,
+				},
+				{
+					conditions:     []weave.Condition{src},
+					msg:            &DistributeMsg{RevenueID: asSeqID(1)},
+					blocksize:      101,
+					wantCheckErr:   nil,
+					wantDeliverErr: nil,
+				},
+			},
+		},
+		"revenue without an account distributing funds": {
+			prepareAccounts: nil,
+			wantAccounts:    nil,
+			actions: []action{
+				{
+					conditions: []weave.Condition{src},
+					msg: &NewRevenueMsg{
+						Admin: []byte("f427d624ed29c1fae0e2"),
+						Recipients: []*Recipient{
+							{Weight: 1, Address: addr1},
+							{Weight: 2, Address: addr2},
+						},
+					},
+					blocksize:      100,
+					wantCheckErr:   nil,
+					wantDeliverErr: nil,
+				},
+				{
+					conditions: []weave.Condition{src},
+					msg: &DistributeMsg{
+						RevenueID: asSeqID(1),
+					},
+					blocksize:      101,
+					wantCheckErr:   nil,
+					wantDeliverErr: nil,
+				},
+			},
+		},
+		"revenue with an account but without enough funds": {
+			prepareAccounts: []account{
+				{address: asSeqID(1), coins: x.Coins{coinp(0, 1, "BTC")}},
+			},
+			wantAccounts: []account{
+				{address: asSeqID(1), coins: x.Coins{coinp(0, 1, "BTC")}},
+			},
+			actions: []action{
+				{
+					conditions: []weave.Condition{src},
+					msg: &NewRevenueMsg{
+						Admin: []byte("f427d624ed29c1fae0e2"),
+						Recipients: []*Recipient{
+							{Weight: 1, Address: addr1},
+							{Weight: 2, Address: addr2},
+						},
+					},
+					blocksize:      100,
+					wantCheckErr:   nil,
+					wantDeliverErr: nil,
+				},
+				{
+					conditions: []weave.Condition{src},
+					msg: &DistributeMsg{
+						RevenueID: asSeqID(1),
+					},
+					blocksize:      101,
+					wantCheckErr:   nil,
+					wantDeliverErr: nil,
+				},
+			},
+		},
+		"distribute revenue with a leftover funds": {
+			prepareAccounts: []account{
+				{address: asSeqID(1), coins: x.Coins{coinp(0, 7, "BTC")}},
+			},
+			wantAccounts: []account{
+				{address: asSeqID(1), coins: x.Coins{coinp(0, 1, "BTC")}},
+				{address: addr1, coins: x.Coins{coinp(0, 2, "BTC")}},
+				{address: addr2, coins: x.Coins{coinp(0, 4, "BTC")}},
+			},
+			actions: []action{
+				{
+					conditions: []weave.Condition{src},
+					msg: &NewRevenueMsg{
+						Admin: []byte("f427d624ed29c1fae0e2"),
+						Recipients: []*Recipient{
+							{Weight: 10000, Address: addr1},
+							{Weight: 20000, Address: addr2},
+						},
+					},
+					blocksize:      100,
+					wantCheckErr:   nil,
+					wantDeliverErr: nil,
+				},
+				{
+					conditions: []weave.Condition{src},
+					msg: &DistributeMsg{
+						RevenueID: asSeqID(1),
+					},
+					blocksize:      101,
+					wantCheckErr:   nil,
+					wantDeliverErr: nil,
+				},
+			},
+		},
+		"distribute revenue with an account holding various tickers": {
```

Comment:
very nice! covering lots of cases

Just two comments here...
Maybe(?) some could be writen cleaner just testing the distribute function, but if it wasn't too much pain to like, I do quite like that you test the full Handler stack that will be executed.

Secondly, please add a test where there is a whole balance, when it cannot be divided evenly,and  has to overflow to fractional (2 recipienents with weight 1 splitting 3 ETH, not just the perfect division below)